### PR TITLE
Automated cherry pick of #61404: apiserver's webhook admission use its own scheme

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Cherry pick of #61404 on release-1.10.

#61404: apiserver's webhook admission use its own scheme

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@SerialVelocity

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -432,12 +460,12 @@ func BuildGenericConfig(s *options.ServerRunOptions, proxyTransport *http.Transp
...
-	storageFactory, err := BuildStorageFactory(s, genericConfig.MergedResourceConfig)
-	if err != nil {
-		return nil, nil, nil, nil, nil, err
+	storageFactory, lastErr := BuildStorageFactory(s, genericConfig.MergedResourceConfig)
+	if lastErr != nil {
+		return
 	}
-	if err := s.Etcd.ApplyWithStorageFactoryTo(storageFactory, genericConfig); err != nil {
-		return nil, nil, nil, nil, nil, err
+	if lastErr = s.Etcd.ApplyWithStorageFactoryTo(storageFactory, genericConfig); lastErr != nil {
```
Comment (sttts) [inline cmd/kube-apiserver/app/server.go]: Without all these lastErr refactorings, the PR would be much much smaller. Would feel better with a minimal patch for the release branch.

Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/server.go]: ok, will do some cleanup

Comment (hzxuzhonghu) [reply cmd/kube-apiserver/app/server.go]: We add a new return value for `CreateKubeAPIServerConfig`, so all the return lines have to be changed even without lastErr

Comment (sttts) [reply cmd/kube-apiserver/app/server.go]: Right. Hmm. For easier pickability, we should maybe leave the commit complete then.

**Comment by MaciekPytel** [comment]
Please add release note to base PR.

Also do we need to cherry-pick #61609? According to https://github.com/kubernetes/kubernetes/pull/61404#issuecomment-378347969 this PR would introduce flaky test without #61609.

@hzxuzhonghu @sttts @jennybuckley

**Comment by MaciekPytel** [comment]
Friendly ping @hzxuzhonghu. We're getting close to 1.10.1 and this PR must merge before Tuesday to make it in.

**Comment by hzxuzhonghu** [comment]
> Also do we need to include #61609 in cherry-pick? According to #61404 (comment) this PR would introduce flaky test without #61609.

Yes, I just notified the cases are run in parallel.

**Pull request closed without merging**